hypervisor/src/hypervisor.rs

```diff
@@ -118,3 +118,13 @@ pub trait Hypervisor: Send + Sync {
     #[cfg(feature = "tdx")]
     fn tdx_capabilities(&self) -> Result<TdxCapabilities>;
 }
+///
+/// Generic MemoryRegion struct
+///
+pub struct MemoryRegion {
```

Comment:
You can use `#[derive(Default)]` so that you don't need to worry about the fields which are not related to mshv while defining this for mshv_memory_region.